packages/apps/dashboard/server/src/modules/details/details.service.ts

```diff
@@ -237,22 +282,56 @@ export class DetailsService {
     }
   }
 
+  /**
+   * Assigns reputations to leaders based on fetched reputation records.
+   *
+   * Logic Description:
+   * - A map of leaders is created using their addresses for quick lookup.
+   * - If ordered by REPUTATION:
+   *   - Iterate through reputation records.
+   *   - If a corresponding leader is found, update their reputation.
+   *   - If no leader is found:
+   *     - Log a warning indicating a missing leader for the specific reputation record.
+   *     - This likely indicates an issue where the subgraph did not return the expected leader data.
+   *     - Skip processing the record without substituting a different leader.
+   *   - Return the top `first` leaders based on reputation.
+   * - If not ordered by REPUTATION:
+   *   - Create a map of reputation records for quick lookup.
+   *   - Update each leader's reputation if found, otherwise set it to LOW.
+   * - Return the list of updated leaders.
+   */
   private assignReputationsToLeaders(
     leaders: LeaderDto[],
     reputations: { address: string; reputation: string }[],
-    chainId: ChainId,
-  ) {
+    orderBy: LeadersOrderBy,
```

Comment:
I think ordering shouldn't be here. As the method name says, here we should just map leaders and their reputations.